### ISSUE TITLE
inaccurate eval loss computation

### ISSUE THREAD

**Issue by wwt17** [huggingface/transformers#41898]
https://github.com/huggingface/transformers/blob/1f0b490a2c42eb129dccc69031ccb537058689c4/examples/pytorch/language-modeling/run_clm_no_trainer.py#L657 The last batch may be smaller, since `drop_last` default to `False` for `DataLoader` construction. Thus,  computing the mean over the losses is inaccurate.
Same issue for the train loss:
https://github.com/huggingface/transformers/blob/1f0b490a2c42eb129dccc69031ccb537058689c4/examples/pytorch/language-modeling/run_clm_no_trainer.py#L673
